src/renderers/webgl/WebGLGeometries.js

```diff
@@ -6,6 +6,20 @@ function WebGLGeometries( gl, attributes, info, bindingStates ) {
 
 	const geometries = new WeakMap();
 	const wireframeAttributes = new WeakMap();
+	const useWeakRef = typeof WeakRef !== 'undefined' && typeof FinalizationRegistry !== 'undefined';
```

Comment:
For the record: At least `FinalizationRegistry` is not supported in IE and Safari.

Reply:
Yeah according to [caniuse](https://caniuse.com/?search=WeakRef) Safari and IE are the two major browsers (if we still consider IE a major browser) that WeakRef is not supported in.

I did find this webkit thread that indicates WeakRef has been implemented and FinalizationRegistry is in progress, as well, if not done -- so it should be coming to Safari at some point:

https://bugs.webkit.org/show_bug.cgi?id=198710

Reply:
Updates to this old thread:
1. [FinalizationRegistry is supported by all browsers](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/FinalizationRegistry#browser_compatibility)
2. [WeakRef is supported by all except Opera](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/WeakRef#browser_compatibility)
3. [IE has been retired on June 15, 2022](https://techcommunity.microsoft.com/t5/windows-it-pro-blog/internet-explorer-11-desktop-app-retirement-faq/ba-p/2366549)

Reply:
@Mugen87 @mrdoob it's been two years since I first proposed this and WeakRefs are now supported in all major browsers. Is there any interest in this? There is currently a memory leak when using multiple renderers on a page. If so I can update the PR.

Reply:
Geometries are not maintained in a weak map anymore. It's a plain object. This should make it easier to implement `WebGLGeometries.dispose()` since you can iterate over `geometries` again.